src/magento/index.js

```diff
@@ -118,30 +85,38 @@ class Magento {
       fetch(uri, { method, headers, body: JSON.stringify(data) })
         .then(response => {
           console.log(response);
-          return response.json();
+          if (response.ok) {
+            return response.json();
+          }
+          // Possible 401 or other network error
+          return response.json().then(errorResponse => Promise.reject(errorResponse));
         })
         .then(responseData => {
-          // TODO: check response code
           // debugger;
           console.log(responseData);
           resolve(responseData);
         })
         .catch(error => {
           console.log(error);
-          reject(error);
+          const customError = this.getErrorMessageForResponce(error);
+          reject(customError);
         });
     });
   }
 
   getErrorMessageForResponce(data) {
     const params = data.parameters;
     let message = data.message;
-    if (typeof params !== 'undefined' && params.length > 0) {
-      data.parameters.forEach((item, index) => {
-        message = message.replace(`%${index + 1}`, item);
-      });
-      return message;
-    }
+    if (typeof params !== 'undefined') {
+      if (Array.isArray(params) && params.length > 0) {
+        data.parameters.forEach((item, index) => {
+          message = message.replace(`%${index + 1}`, item);
+        });
+        return message;
+      } else if (params.resources !== undefined) {
```

Comment:
When you provide wrong integration token, and try to access `/V1/store/storeConfigs` url, the response code is 401(which is correct) and the response contain `parameters` which should be of type Array but comes out to be an object, because of that this check was included.